gui/src/components/mainInput/ContinueInputBox.tsx

```diff
@@ -112,26 +112,28 @@ interface ContinueInputBoxProps {
   source?: "perplexity" | "aider" | "continue";
 }
 
+
 const ContinueInputBox = memo(function ContinueInputBox({
   isLastUserInput,
   isMainInput,
   onEnter,
   editorState,
-  contextItems,
   hidden,
   source = "continue",
+  contextItems = [],
 }: ContinueInputBoxProps) {
   const dispatch = useDispatch();
 
   const active = useSelector((store: RootState) => {
-    switch (source) {
-      case "perplexity":
-        return store.state.perplexityActive;
-      case "aider":
-        return store.state.aiderActive;
-      default:
-        return store.state.active;
-    }
+    const state = store?.state;
+
+    if (!state) return false;
+
+    return source === "perplexity"
+      ? state.perplexityActive
+      : source === "aider"
+      ? state.aiderActive
+      : state.active;
```

Comment:
The reason I had changed this is because it was adding to unnecessary re-renders when there was errors.  This was a refactor for helping with functionality, not just because I didn't like the switch statement.  This aided with fixing the error issue.

Reply:
Done